dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/ProductViewTest.php

```diff
@@ -670,6 +670,30 @@ public function testProductPrices()
         $this->assertBaseFields($secondProduct, $response['products']['items'][1]);
     }
 
+    /**
+     * @magentoApiDataFixture Magento/Catalog/_files/products_for_search.php
+     */
+    public function testProductFilterNfinset()
+    {
+        $productSku = 'search_product_1';
+
+        $query = <<<QUERY
+       {
+           products(filter: {sku: {nfinset: ["$productSku"]}})
+           {
+               items {
+                   id
+                  name
+                  sku
+               }
+           }
+       }
+QUERY;
+
+        $response = $this->graphQlQuery($query);
+        $this->assertNotEquals($response['products']['items'][0]['sku'], $productSku);
```

Comment:
Need to make sure that product is missed in full result
`self::assertNotContains($productSku, array_column($items, 'sku'))`